mf.cpp

```diff
@@ -976,7 +976,7 @@ void Utility::grid_shuffle_scale_problem_on_disk(
 mf_float* Utility::malloc_aligned_float(mf_long size)
 {
     // Check if conversion from mf_long to size_t causes overflow.
-    if (size > numeric_limits<std::size_t>::max() / sizeof(mf_float) + 1)
+    if ((unsigned long)size > numeric_limits<std::size_t>::max() / sizeof(mf_float) + 1)
```

Comment:
check if `sizeof(mf_long) <= sizeof(unsigned long)` before applying this transform.

Reply:
Hey @wschin ,
On MacOS, `sizeof(mf_long) = 8` and `sizeof(unsigned long) = 8`.
On Windows, `sizeof(mf_long) = 8` and  `sizeof(unsigned long) = 4`.
In my experience, `size` is always greater than 0 with the ML .NET unit test `MatrixFactorizationSimpleTrainAndPredict()`.

Reply:
Yes, your fix is correct for the current codebase. But `mf_long` is a user-defined type. There is no guarantee that `mf_long` will be `long long` forever.

Reply:
Ok, great! I have added the additional size checks to the if statement. Thank you!